lib/utils/packaging/unarchive.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package packaging
+
+import (
+	"archive/zip"
+	"io"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+const (
+	reservedFreeDisk = 10 * 1024 * 1024
+)
+
+// RemoveWithSuffix removes all files in dir that have the provided suffix, except for files named `skipName`
+func RemoveWithSuffix(extractDir, suffix, skipName string) error {
```

Comment:
```suggestion
func RemoveWithSuffix(dir, suffix, skipName string) error {
```
Nothing is being extracted?